zerver/lib/export.py

```diff
@@ -1528,3 +1519,46 @@ def export_messages_single_user(user_profile: UserProfile, output_dir: Path,
         write_message_export(message_filename, message_output)
         min_id = max(user_message_ids)
         dump_file_id += 1
+
+def export_analytics_tables(realm: Realm, output_dir: Path) -> None:
+    response = {}  # type: TableData
+
+    export_file = os.path.join(output_dir, "analytics.json")
+    logging.info("Writing analytics table data to %s", (export_file))
+    config = get_analytics_config()
+    export_from_config(
+        response=response,
+        config=config,
+        seed_object=realm,
+    )
+    write_data_to_file(output_file=export_file, data=response)
+
+def get_analytics_config() -> Config:
+
+    analytics_config = Config(
+        table='zerver_analytics',
+        is_seeded=True,
+    )
+
+    Config(
+        table='analytics_realmcount',
+        model=RealmCount,
+        normal_parent=analytics_config,
+        parent_key='realm_id__in',
+    )
+
+    Config(
+        table='analytics_usercount',
+        model=UserCount,
+        normal_parent=analytics_config,
+        parent_key='realm_id__in',
+    )
+
+    Config(
+        table='analytics_streamcount',
+        model=StreamCount,
+        normal_parent=analytics_config,
+        parent_key='realm_id__in',
+    )
+
+    return analytics_config
```

Comment:
@whoodes what I had been imagining is doing something more like, just before we write things out to disk, doing this:
analytics_data = {(key, value) for key, value in response if key in ANALYTICS_TABLES}
response = {(key, value) for key, value in response if key not in ANALYTICS_TABLES}

Is there a reason you didn't go with that approach?

In any case, you should also update the import tool to read the new file as well.

Reply:
> Is there a reason you didn't go with that approach?

@timabbott Just from reading through the export file. I made the assumption that the refactor should modularize the analytics_data as well.  Next time, I will inquire first before implementing assumptions I have.  If there's no real benefit to this approach, I'll readjust asap.  Thank you.

Reply:
@whoodes I think this approach is likely cleaner; I'm just not 100% sure it's correct.  Can you do the manual test of doing both export approaches and comparing the resulting `analytics.json` files?  They should be identical.  This is a good way to end up confident in this change.

Reply:
(And the other thing I'll want to do before manually testing and merging this is having the import side done)

Reply:
Sounds good @timabbott. I'll work on doing those comparative tests, as well as updating the import tool.  Would this be an appropriate situation to mark this PR as [WIP]?

Reply:
No need -- just post a comment when you're ready for another review.  We manage our review queue through GitHub notifications, so WIP is primarily useful when first opening a PR.